mule-extensions-api-xml-dsl/src/main/java/org/mule/runtime/extension/xml/dsl/api/property/XmlElementStyleModelProperty.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.xml.dsl.api.property;
+
+import org.mule.runtime.extension.api.annotation.dsl.xml.XmlElementStyle;
+import org.mule.runtime.extension.api.introspection.ModelProperty;
+
+/**
+ * Contains directives regarding syntax and semantics of the generated XML DSL
+ *
+ * @since 1.0
+ */
+public class XmlElementStyleModelProperty implements ModelProperty
```

Comment:
I don't know if `style` is the best name for this, since `allowReferences` has nothing to do with the Style of the xml but how it is interpreted. 
Maybe `XmlElementProperties`?